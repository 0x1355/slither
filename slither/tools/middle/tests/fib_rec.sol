pragma solidity ^0.6.4;

contract Fib {
    function fib(uint n) public returns(uint res) {
        uint ret;
        if (n <= 1) {
            ret = n;
        } else {
            ret = Fib.fib(n - 1) + Fib.fib(n - 2);
        }
        return ret;
    }
}