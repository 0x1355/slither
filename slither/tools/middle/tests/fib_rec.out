OVERLAY_FUNCTION_
    ENTRY_POINT
    EXPRESSION
OVERLAY_FUNCTION_
    ENTRY_POINT
    EXPRESSION
fib
    ENTRY_POINT
    NEW VARIABLE
    IF
    OVERLAY
    OVERLAY
    END_IF
    RETURN
